pkg/agent/endpoints/sdsv3/handler_test.go

```diff
@@ -500,7 +493,7 @@ func TestStreamSecrets(t *testing.T) {
 					},
 				},
 			},
-			expectSecrets: []*tls_v3.Secret{workloadTLSCertificate3},
+			expectSecrets: []*tls_v3.Secret{tdValidationContext},
```

Comment:
Woof. I'm glad you caught that these new tests weren't testing the validation context!